File: 383-ransom-note/383-ransom-note.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        
        freq = {}
        
        for i in ransomNote:
            if i not in freq.keys():
                freq[i] = 1
            else:
                freq[i] += 1
                
        for i in magazine:
            if i in freq.keys(): freq[i] -= 1
            
        for i in freq.keys():
            if freq[i]>0:
                return False
        return True;
        

File: 383-ransom-note/383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<int,int> freq;
        
        for(auto i:magazine){
            freq[i]++;
        }
        
        for(auto i:ransomNote){
            if(!freq[i]) return false;
            freq[i]--;
        }
        return true;
        
        
    }
};